### ISSUE TITLE
test-containers: the graph lane pins PostgreSQL 19 without pgvector, so the graph-storage schema cannot be created

### ISSUE THREAD

**Issue by vasylcf** [constructorfabric/gears-rust#4666]
### What happens

`cf_gears_test_containers::postgres_graph()` pins `postgres:19beta3-alpine`. That image has SQL/PGQ but no `pgvector`, so the graph-storage gear's first migration statement fails:

```
CREATE EXTENSION IF NOT EXISTS vector SCHEMA public;
-- ERROR 0A000: extension "vector" is not available
```

The whole schema migration aborts, so the gear's PostgreSQL 19 test lane cannot create a single table — this is not "vector search is unavailable", it is "the suite never reaches the tests".

### Why it was not caught when the pin landed

The pin was added in #4616 ahead of its consumer, with a comment saying so: *"reserved for the graph-storage gear… Unused until that gear lands."*

Until now its only consumer has been `libs/toolkit-db/tests/pg/secure_graph.rs`, whose fixtures are plain tables — the word `vector` does not appear in that file. Those tests are green on the stock image, so the pin looked correct.

graph-storage is the second consumer and the first that needs **both** capabilities in one schema: `GRAPH_TABLE` for the traversal backend (PostgreSQL 19+) and `pgvector` for the `embedding VECTOR(384)` column and its HNSW index. ADR-0001 makes "PostgreSQL 16 or later **with pgvector**" the gear's baseline and PG19 the probed capability on top of it.

### Current workaround, and why it is not enough

The gear's lane reads `GEARS_TEST_PG_GRAPH_IMAGE` for an image carrying both, skips with a named reason when none is configured, and fails loudly under `GEARS_TEST_PG_GRAPH_REQUIRED=1` so a CI lane cannot report success having run nothing.

That is correct but per-developer: a fresh clone silently skips the lane, because the image only exists on machines that happen to have built one. It also cannot be made a required CI check in this state.

### Proposed fix

Publish one image with both capabilities and have `postgres_graph()` return it, so the lane works from a fresh clone with no environment variable.

The recipe already exists and is in daily use — `studio-web:studio-backend/docker/graph-postgres/Dockerfile` builds it, and a sibling `verify-cnpg-compatibility.sh` exercises the operand contract plus the two database features the gear needs:

```dockerfile
# CloudNativePG operand, pinned by multi-arch digest (CNPG replaces the
# entrypoint with its instance manager, so this is the contract we depend on
# rather than a Docker Official Image).
ARG CNPG_POSTGRES_IMAGE=ghcr.io/cloudnative-pg/postgresql:19beta3-standard-trixie@sha256:066b9e2245e5c354a57b53d71a4517bfe203aabbe7286778981b98e177ee79c3
FROM ${CNPG_POSTGRES_IMAGE} AS pgvector-build

# No released pgvector targets PG19 yet — upstream added support on master
# (pgvector#1005, closed 2026-07-29). Pin the commit so builds reproduce.
ARG PGVECTOR_REF=5219575

USER root
RUN set -eux; \
    apt-get update; \
    apt-get install -y --no-install-recommends \
        build-essential ca-certificates git postgresql-server-dev-19; \
    git clone https://github.com/pgvector/pgvector.git /tmp/pgvector; \
    cd /tmp/pgvector; \
    git checkout "${PGVECTOR_REF}"; \
    make OPTFLAGS=""; \
    make install; \
    rm -rf /tmp/pgvector /var/lib/apt/lists/*

FROM ${CNPG_POSTGRES_IMAGE}
COPY --from=pgvector-build /usr/share/postgresql/19/extension/vector* /usr/share/postgresql/19/extension/
COPY --from=pgvector-build /usr/lib/postgresql/19/lib/vector.so /usr/lib/postgresql/19/lib/
```

Two details worth carrying over rather than rediscovering: the base is pinned **by digest**, not by tag, because the tag moves during beta; and `pgvector` is built from source with `OPTFLAGS=""`, which is what upstream requires for a portable build.

If a published image is not wanted while PG19 is pre-GA, the alternative is to keep the environment-variable path but say so in `libs/test-containers`, so the next gear author does not have to find it by watching a migration fail.

### What this closes by itself

PostgreSQL 19 reaches GA around October 2026, after which pgvector ships normal releases for it and a stock image with the extension is enough. So this is temporary — but "temporary" here is a couple of months, and the graph lane is wanted as a required check sooner than that.

### Related

- #4616 — where the pin was introduced, ahead of its consumer
- #4639 — the SQL/PGQ layer whose tests are the pin's first (and pgvector-free) consumer
- ADR-0001 in #4523 — the PG16+/pgvector baseline, and the PG16-and-PG19 CI matrix this affects. Note the PG16 half is separately unavailable, for the opposite reason: no `GRAPH_TABLE` there at all, so that half only ever exercises the fallback backend.

**Comment by vasylcf** [comment]
Three things this issue got wrong or has outgrown, learned by building the
image it asks for and running the lane on it.

**The recipe above does not work for `test-containers`.** The CloudNativePG
operand replaces the entrypoint with its instance manager, and
`test-containers` starts a server by passing `-c …` arguments to the image's
own `docker-entrypoint.sh`. With the operand base the container exits
immediately with `exec: "-c": executable file not found`, before any
migration runs. The CNPG contract is the right one for a *deployment* — which
is why the studio assembly builds it that way — and the wrong one for this
lane. The official `postgres:19beta3` base works, keeps the stock entrypoint
and initdb behaviour, and takes the same two-stage pgvector build.

The working file is in #4794 at
`gears/graph-storage/docker/pg19-pgvector.Dockerfile`, pinned to the same
pgvector revision. It is not a sketch: CI builds it and runs the gear's whole
PostgreSQL lane on it, 66 cases plus the coverage floor, in under five
minutes on a stock runner.

**"It cannot be made a required CI check in this state" is out of date.** It
can, and in #4794 it is — by having the *consumer* build the image in its own
job. So the lane is no longer blocked, and what remains of this issue is the
narrower thing: every gear that needs both capabilities has to carry its own
image and its own build step, and for those gears the pin has stopped being a
pin. That is a duplication problem now rather than a blocked-lane one, which
is a fair reason to leave it until GA.

**The escape hatch still has no signpost.** Whatever is decided about
publishing an image, the one cheap thing worth doing now is the alternative
this issue already names: say in `libs/test-containers` that `postgres_graph()`
carries no pgvector, and what to do about it. Today a gear author finds out by
watching `CREATE EXTENSION vector` fail.

**What closes it.** Still PostgreSQL 19 GA, and the date has not moved:
pgvector publishes no PG19 image yet (Docker Hub tops out at `0.8.6-pg18` as
of today). When it does, this collapses to a one-line change of
`POSTGRES_GRAPH_TAG` plus deleting the Dockerfile and the build step from the
graph-storage job.
